### PR TITLE
Bump to version 1328, remove Vararg testing patch, degit patches, Conda.jl

### PR DESCRIPTION
**Pull request by mkitti**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mkitti** [comment]
@conda-forge-admin, please rerender

**Comment by mkitti** [comment]
Fixes #161 via c60af33, requires Conda.jl pull request https://github.com/JuliaPy/Conda.jl/pull/216

File: recipe/patches/0002-disable-testing-Baz.baz-1.patch [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-
+#From 31e9de0137db669f1a41e933da701b0e2c7ea6f2 Mon Sep 17 00:00:00 2001
+#From: ngam <ngam@users.noreply.github.com>
+#Date: Mon, 27 Dec 2021 01:36:51 +0000
+#Subject: [PATCH 2/3] disable testing Baz.baz == 1
+#
+#---
+# test/precompile.jl | 2 +-
+# 1 file changed, 1 insertion(+), 1 deletion(-)
+#
```
Comment (ngam) [inline recipe/patches/0002-disable-testing-Baz.baz-1.patch:9]:
```suggestion
```
Let's just remove this?

File: recipe/patches/0002-disable-testing-Baz.baz-1.patch [hunk 2]
```diff
@@ -20,6 +20,6 @@ index 51b25b8325..8abd491a63 100644
  
      # Issue #12720
      FooBar1_file = joinpath(dir, "FooBar1.jl")
--- 
-2.34.1
+#--
```
Comment (ngam) [inline recipe/patches/0002-disable-testing-Baz.baz-1.patch:23]:
```suggestion
```

Remove

File: recipe/patches/0002-disable-testing-Baz.baz-1.patch [hunk 3]
```diff
@@ -20,6 +20,6 @@ index 51b25b8325..8abd491a63 100644
  
      # Issue #12720
      FooBar1_file = joinpath(dir, "FooBar1.jl")
--- 
-2.34.1
+#-- 
+#2.34.1
```
Comment (ngam) [inline recipe/patches/0002-disable-testing-Baz.baz-1.patch:24]:
```suggestion
```

remove

File: recipe/patches/julia-hardcoded-libs.patch [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
+#
+#---
+# stdlib/LibGit2_jll/src/LibGit2_jll.jl   |  4 ++--
+# stdlib/LibSSH2_jll/src/LibSSH2_jll.jl   |  4 ++--
+# stdlib/MbedTLS_jll/src/MbedTLS_jll.jl   | 12 ++++++------
+# stdlib/OpenLibm_jll/src/OpenLibm_jll.jl |  4 ++--
+# stdlib/libLLVM_jll/src/libLLVM_jll.jl   |  2 +-
+# stdlib/nghttp2_jll/src/nghttp2_jll.jl   |  4 ++--
+# 6 files changed, 15 insertions(+), 15 deletions(-)
+#
```
Comment (ngam) [inline recipe/patches/julia-hardcoded-libs.patch:14]:
```suggestion
```

remove. Could you also fix the LLVM part?

File: recipe/patches/julia-hardcoded-libs.patch [hunk 5]
```diff
@@ -113,6 +113,6 @@ index 8b98c76..d926449 100644
  end
  
  function __init__()
--- 
-2.33.0
+#-- 
+#2.33.0
```
Comment (ngam) [inline recipe/patches/julia-hardcoded-libs.patch:117]:
```suggestion
```

remove

**Review by ngam — CHANGES_REQUESTED** [review]

File: recipe/patches/julia-hardcoded-libs.patch (hunk 4 above)
Comment (ngam) [reply recipe/patches/julia-hardcoded-libs.patch:14]: We still use the vendored LLVM, so please delete this part:

```diff
diff --git a/stdlib/libLLVM_jll/src/libLLVM_jll.jl b/stdlib/libLLVM_jll/src/libLLVM_jll.jl
index fa45e75..b51812e 100644
--- a/stdlib/libLLVM_jll/src/libLLVM_jll.jl
+++ b/stdlib/libLLVM_jll/src/libLLVM_jll.jl
@@ -23,7 +23,7 @@ if Sys.iswindows()
 elseif Sys.isapple()
     const libLLVM = "@rpath/libLLVM.dylib"
 else
-    const libLLVM = "libLLVM-13jl.so"
+    const libLLVM = "libLLVM.so"
 end
 
 function __init__()
```

Comment (mkitti) [reply recipe/patches/julia-hardcoded-libs.patch:14]: I didn't change this part.

Comment (ngam) [reply recipe/patches/julia-hardcoded-libs.patch:14]: I know, I am suggesting we delete this part completely from the file.

I also think we should split this patch into sub-patches. Do you want me to do that? I just think doing all the libraries at once is a recipe for disaster --- we will miss some inconsistencies; instead, we should do a patch per library. For example, both libgit2 and mbedtls will need to be os-specific due to our problems with osx.

Comment (ngam) [reply recipe/patches/julia-hardcoded-libs.patch:14]: So this likely introduces problems that we are not detecting. If you follow the log line per line (which I did), I did notice it was raising some warnings about LLVM-13jl and that immediately made me think of this particular patch and how it made no sense as we are using their vendored LLVM. 

Does that make sense to you?

Comment (mkitti) [reply recipe/patches/julia-hardcoded-libs.patch:14]: These are just symlinks to each other `libLLVM.so` is a symlink to `libLLVM-13jl.so`.

**Comment by mkitti** [comment]
If you want to mess with the patches now, that's fine with me. I'm not sure if the commenting out had the desired effect, or not. The desired effect would be for the reported build number via `versioninfo()` to match the version on the package.

**Comment by ngam** [comment]
> If you want to mess with the patches now, that's fine with me. I'm not sure if the commenting out had the desired effect, or not. The desired effect would be for the reported build number via `versioninfo()` to match the version on the package.

Let's add a test or something for that --- `julia -e "versioninfo()"?

File: recipe/patches/julia-hardcoded-libs.patch (hunk 4 above)
Comment (ngam) [reply recipe/patches/julia-hardcoded-libs.patch:14]: Yes, but I'd rather not mess with it if we are not using our own LLVM. In other words, I would rather use whatever they have upstream.

**Comment by ngam** [comment]
```
~$ julia                    
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.8.0-DEV.1262 (2021-09-20)
 _/ |\__'_|_|_|\__'_|  |  HEAD/9ea2cc48b* (fork: 1 commits, 6 days)
|__/                   |

julia> versioninfo()
Julia Version 1.8.0-DEV.1262
Commit 9ea2cc48b* (2021-09-20 14:56 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin13.4.0)
  CPU: Apple M1 Max
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.0 (ORCJIT, westmere)
Environment:
  JULIA_DEPOT_PATH_BACKUP = 
  JULIA_PROJECT_BACKUP = 
  JULIA_LOAD_PATH_BACKUP = 
  JULIA_DEPOT_PATH = /Users/ngam/.Mambaforge-MacOSX-x86_64/envs/jlia/share/julia:
  JULIA_PROJECT = @jlia
  JULIA_LOAD_PATH = @:@jlia:@stdlib
  JULIA_SSL_CA_ROOTS_PATH_BACKUP = 
  JULIA_SSL_CA_ROOTS_PATH = /Users/ngam/.Mambaforge-MacOSX-x86_64/envs/jlia/ssl/cacert.pem

julia> 
```

```
julia -e 'using InteractiveUtils; InteractiveUtils.versioninfo()'
Julia Version 1.8.0-DEV.1262
Commit 9ea2cc48b* (2021-09-20 14:56 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin13.4.0)
  CPU: Apple M1 Max
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.0 (ORCJIT, westmere)
Environment:
  JULIA_DEPOT_PATH_BACKUP = 
  JULIA_PROJECT_BACKUP = 
  JULIA_LOAD_PATH_BACKUP = 
  JULIA_DEPOT_PATH = /Users/ngam/.Mambaforge-MacOSX-x86_64/envs/jlia/share/julia:
  JULIA_PROJECT = @jlia
  JULIA_LOAD_PATH = @:@jlia:@stdlib
  JULIA_SSL_CA_ROOTS_PATH_BACKUP = 
  JULIA_SSL_CA_ROOTS_PATH = /Users/ngam/.Mambaforge-MacOSX-x86_64/envs/jlia/ssl/cacert.pem
```

**Comment by ngam** [comment]
> If you want to mess with the patches now, that's fine with me

No urgency to do this now, I can do it in another PR if you want this to go ahead sooner

**Comment by ngam** [comment]
> Is it the `1261` vs `1262`? That's fine

What I really want us to do is to rename julia to julia-cf for these experimental releases and keep redistribute a wholesome (as close to upstream as possible) julia

**Comment by ngam** [comment]
```
    julia:             1.8.0.dev.1328-h0563ae0_0  local      

```

```
+ julia -e 'using InteractiveUtils; InteractiveUtils.versioninfo()'
Julia Version 1.8.0-DEV.1328
Commit 352266160* (2022-01-16 06:03 UTC)
Platform Info:
  OS: Linux (x86_64-conda-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU E5-2673 v4 @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.0 (ORCJIT, broadwell)
Environment:
  JULIA_SSL_CA_ROOTS_PATH = $PREFIX/ssl/cacert.pem
  JULIA_DEPOT_PATH = $PREFIX/share/julia:
  JULIA_PROJECT = @_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_
  LD_RUN_PATH = $PREFIX/lib
  JULIA_PROJECT_BACKUP = 
  JULIA_LOAD_PATH = @:@_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_:@stdlib
  JULIA_DEPOT_PATH_BACKUP = 
  JULIA_LOAD_PATH_BACKUP = 
  JULIA_SSL_CA_ROOTS_PATH_BACKUP = 

```

File: recipe/scripts/activate.sh [hunk 6]
```diff
@@ -16,3 +16,10 @@ export JULIA_LOAD_PATH="@:$JULIA_PROJECT:@stdlib"
 # make julia use same cert
  export JULIA_SSL_CA_ROOTS_PATH_BACKUP=${JULIA_SSL_CA_ROOTS_PATH:-}
  export JULIA_SSL_CA_ROOTS_PATH=$CONDA_PREFIX/ssl/cacert.pem
+
+# Setup variables for Conda.jl
+conda_path=$(which conda)
```
Comment (ngam) [inline recipe/scripts/activate.sh:21]: double check if this actually works as I am not seeing it in the logs so far.

We may be able to hard-wire this.

```
ngam@mp15:~$ uname -a
Darwin mb15 19.6.0 Darwin Kernel Version 19.6.0: Sun Nov 14 19:58:51 PST 2021; root:xnu-6153.141.50~1/RELEASE_X86_64 x86_64
ngam@mp15:~$ which conda
conda () {
	\local cmd="${1-__missing__}"
	case "$cmd" in
		(activate | deactivate) __conda_activate "$@" ;;
		(install | update | upgrade | remove | uninstall) __conda_exe "$@" || \return
			__conda_reactivate ;;
		(*) __conda_exe "$@" ;;
	esac
}

```

```
ngam@t450:~$ uname -a
Linux t450 5.11.0-46-generic #51~20.04.1-Ubuntu SMP Fri Jan 7 06:51:40 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
ngam@t450:~$ which conda
/home/ngam/.Mambaforge-Linux-x86_64/condabin/conda

```

```
~$ uname -a
Darwin mp14 21.2.0 Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:41 PST 2021; root:xnu-8019.61.5~1/RELEASE_ARM64_T6000 arm64
~$ which conda
conda () {
	\local cmd="${1-__missing__}"
	case "$cmd" in
		(activate | deactivate) __conda_activate "$@" ;;
		(install | update | upgrade | remove | uninstall) __conda_exe "$@" || \return
			__conda_reactivate ;;
		(*) __conda_exe "$@" ;;
	esac
}
~$ 

```

Comment (ngam) [reply recipe/scripts/activate.sh:21]: The above is due to zsh --- I am not sure if the osx version would be okay given that osx now defaults to zsh. Need an explicit instruction to use bash somewhere?

Comment (mkitti) [reply recipe/scripts/activate.sh:21]: How about `$CONDA_EXE`?

Comment (ngam) [reply recipe/scripts/activate.sh:21]: > How about `$CONDA_EXE`?

This would be best, I think

**Review by ngam — APPROVED** [review]

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

**Comment by ngam** [comment]
Merging, thanks @mkitti.

**Comment by mkitti** [comment]
Did you forget to merge?

**Comment by ngam** [comment]
Wanted to give you a chance to see it :)

**Pull request merged**